pkg/ccl/changefeedccl/changefeedbase/options.go

```diff
@@ -943,21 +943,26 @@ func (e EncodingOptions) Validate() error {
 		return errors.Errorf(`%s is only usable with %s=%s/%s`, OptHeadersJSONColumnName, OptFormat, OptFormatJSON, OptFormatAvro)
 	}
 
-	if e.Envelope != OptEnvelopeWrapped && e.Format != OptFormatJSON && e.Format != OptFormatParquet {
+	// TODO(#140110): refactor this logic.
+	if (e.Envelope != OptEnvelopeWrapped && e.Envelope != OptEnvelopeEnriched) && e.Format != OptFormatJSON && e.Format != OptFormatParquet {
 		requiresWrap := []struct {
 			k string
 			b bool
 		}{
 			{OptKeyInValue, e.KeyInValue},
+			// NOTE: topic_in_value is allowed for envelope=enriched, but has no
+			// effect. This is because the enriched envelope already has much of
+			// the information contained in the topic (ie table name), but this
+			// option is required for the webhook sink so we must permit it.
 			{OptTopicInValue, e.TopicInValue},
 			{OptUpdatedTimestamps, e.UpdatedTimestamps},
 			{OptMVCCTimestamps, e.MVCCTimestamps},
 			{OptDiff, e.Diff},
 		}
 		for _, v := range requiresWrap {
 			if v.b {
-				return errors.Errorf(`%s is only usable with %s=%s`,
-					v.k, OptEnvelope, OptEnvelopeWrapped)
+				return errors.Errorf(`%s is only usable with %s=%s or %s`,
```

Comment:
nit:
```suggestion
				return errors.Errorf(`%s is only usable with "%[2]s=%s or %[2]s=%s"`,
```

Reply:
what does %[2]s do?

Reply:
yeah you can, i think you can think of each raw `%s` as if it was the next `%[n]s` so
```
`%s is only usable with "%[2]s=%s or %[2]s=%s"`
```
is equivalent to
```
`%[1]s is only usable with "%[2]s=%[3]s or %[2]s=%[4]s"`,
```